### PR TITLE
librats: code optimization and bug fixes

### PR DESCRIPTION
**Pull request by zeuson0**
Signed-off-by: wangya <wangya.zs@alibaba-inc.com>

### CODE REVIEW

File: attesters/sev-snp/collect_evidence.c [hunk 1]
```diff
@@ -15,13 +15,11 @@
...
 #include <librats/log.h>
-#include <curl/curl.h>
 #include "sev_snp.h"
+#include "utils.h"
 
 #define SEV_GUEST_DEVICE "/dev/sev-guest"
-#define KDS_CERT_SITE	 "https://kdsintf.amd.com"
-#define KDS_VCEK	 KDS_CERT_SITE "/vcek/v1/"
-#define CURL_RETRY_TIMES 5
+
```
Comment (haosanzi) [inline attesters/sev-snp/collect_evidence.c:22]: please remove this empty line

File: verifiers/sev-snp/utils.h [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 #include <stdbool.h>
 #include <stdint.h>
-
+#include "../../attesters/sev-snp/utils.h"
```
Comment (haosanzi) [inline verifiers/sev-snp/utils.h:12]: please add an empty line

File: verifiers/sev-snp/verify_evidence.c [hunk 3]
```diff
@@ -29,7 +29,12 @@ rats_verifier_err_t sev_snp_verify_evidence(rats_verifier_ctx_t *ctx,
 	rats_verifier_err_t err = RATS_VERIFIER_ERR_UNKNOWN;
 	snp_attestation_report_t *report = (snp_attestation_report_t *)(evidence->snp.report);
 	if (evidence->snp.vcek[0] == '\0') {
-		return RATS_VERIFIER_ERR_INVALID;
+		memset(evidence->snp.vcek, 0, VECK_MAX_SIZE);
+		err = sev_snp_get_vcek_der(report->chip_id, sizeof(report->chip_id),
+					   &report->platform_version, &evidence->snp);
+		if (err != RATS_ATTESTER_ERR_NONE) {
```
Comment (haosanzi) [inline verifiers/sev-snp/verify_evidence.c:35]: if (err != RATS_ATTESTER_ERR_NONE)
			return err;

File: attesters/tdx-ecdsa/CMakeLists.txt [hunk 4]
```diff
@@ -33,6 +31,7 @@ set(SOURCES cleanup.c
     add_library(${PROJECT_NAME} SHARED ${SOURCES})
     target_link_libraries(${PROJECT_NAME} ${EXTRA_LINK_LIBRARY} ${RATS_LDFLAGS} ${RATS_LIB})
     set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${VERSION} SOVERSION ${VERSION_MAJOR})
-    # Install library
-    install(TARGETS ${PROJECT_NAME}
-        DESTINATION ${RATS_INSTALL_LIBA_PATH})
+
+# Install library
+install(TARGETS ${PROJECT_NAME}
```
Comment (haosanzi) [inline attesters/tdx-ecdsa/CMakeLists.txt:36]: please do not change this format

File: attesters/sev-snp/CMakeLists.txt [hunk 5]
```diff
@@ -22,6 +22,7 @@ set(SOURCES cleanup.c
             init.c
             main.c
             pre_init.c
+            utils.h
```
Comment (haosanzi) [inline attesters/sev-snp/CMakeLists.txt:25]: hi, why add `utils.h` in here?

**Review by haosanzi — APPROVED** [review]

**Review by YangLiang3 — APPROVED** [review]

**Pull request merged**